### PR TITLE
thrimbletrimmer: Add support for advanced crop options

### PR DESCRIPTION
**Pull request by ekimekim**
- Advanced options are in an expandable section
- On load + change of template, set from template defaults
- On load, set from video if not null
- Show only in template mode
- Use when previewing image
- Send when submitting

### CODE REVIEW

**Review by chrusher — APPROVED** [review]
Looks good to me.

File: thrimbletrimmer/edit.html [hunk 1]
```diff
@@ -252,6 +252,31 @@
...
+							<br/>
+
+							Location:
+							<input type="text" class="video-info-thumbnail-position" id="video-info-thumbnail-location-0" />
+							<input type="text" class="video-info-thumbnail-position" id="video-info-thumbnail-location-1" />
+							to
+							<input type="text" class="video-info-thumbnail-position" id="video-info-thumbnail-location-2" />
+							<input type="text" class="video-info-thumbnail-position" id="video-info-thumbnail-location-3" />
+							<br/>
+
```
Comment (ElementalAlchemist) [inline thrimbletrimmer/edit.html:278]: Did you intentionally miss the closing `</details>` tag?

File: thrimbletrimmer/scripts/edit.js [hunk 2]
```diff
@@ -267,27 +271,50 @@ window.addEventListener("DOMContentLoaded", async (event) => {
...
 			const imageTemplate = document.getElementById("video-info-thumbnail-template").value;
-			imageElement.src = `/thumbnail/${globalStreamName}/source.png?timestamp=${imageTime}&template=${imageTemplate}`;
+			const [crop, loc] = getTemplatePosition();
+			const queryParts = [
+				`timestamp=${imageTime}`,
+				`template=${imageTemplate}`,
+				`crop=${crop.join(",")}`,
+				`location=${loc.join(",")}`,
+			];
+			imageElement.src = `/thumbnail/${globalStreamName}/source.png?${queryParts.join("&")}`;
```
Comment (ElementalAlchemist) [inline thrimbletrimmer/scripts/edit.js:281]: We should make sure we document somewhere that template names may not contain ampersands, unless we intend on URI-escaping them.

**Review by ElementalAlchemist — CHANGES_REQUESTED** [review]

File: thrimbletrimmer/scripts/edit.js (hunk 2 above)
Comment (chrusher) [reply thrimbletrimmer/scripts/edit.js:281]: We can enforce template names being limited to `[A-Za-z0-9_-]` when creating or modifying them by adding a check to `thrimshim.validate_template`.

File: thrimbletrimmer/edit.html (hunk 1 above)
Comment (ekimekim) [reply thrimbletrimmer/edit.html:278]: fixed

File: thrimbletrimmer/scripts/edit.js (hunk 2 above)
Comment (ekimekim) [reply thrimbletrimmer/scripts/edit.js]: I've replaced all ad-hoc query string building with proper usage of `URLSearchParams()` which will automatically do correct encoding.

**Pull request merged**